src/s2wasm.h

```diff
@@ -55,8 +56,8 @@ class S2WasmBuilder {
         {}
 
   void build(LinkerObject *obj, LinkerObject::SymbolInfo* info) {
-    if (!obj->isEmpty()) Fatal() << "Cannot construct an S2WasmBuilder in an non-empty LinkerObject";
-    if (!info) info = getSymbolInfo();
+    if (!symbolInfo)
+      symbolInfo.reset(getSymbolInfo());
```

Comment:
what does this mean? perhaps add a comment. in what cases would that var be false and hit this path?